src/init/AppRunner.ts

```diff
@@ -33,41 +35,65 @@ export class AppRunner {
+   * E.g.: `{ rootFilePath: '.data' }`.
+   * Shorthand parameter names can not be used, so `{ f: '.data' }` would not work.
+   *
+   * If a certain would set a specific variable,
+   * and both are present in `variableBindings` and `settings`,
+   * the value in `variableBindings` will take priority.
+   *
    * @param loaderProperties - Components.js loader properties.
    * @param configFile - Path to the server config file.
-   * @param variableBindings - Parameters to pass into the VariableResolver.
+   * @param variableBindings - Bindings of Components.js variables.
+   * @param settings - Settings that need to be resolved.
    */
   public async run(
     loaderProperties: IComponentsManagerBuilderOptions<App>,
     configFile: string,
-    variableBindings: VariableBindings,
+    variableBindings?: VariableBindings,
+    settings?: Settings,
   ): Promise<void> {
-    const app = await this.create(loaderProperties, configFile, variableBindings);
+    const app = await this.create(loaderProperties, configFile, variableBindings, settings);
     await app.start();
   }
 
   /**
    * Returns an App object, created with the given config, that can start and stop the Solid server.
+   *
    * Keys of the `variableBindings` object should be Components.js variables.
    * E.g.: `{ 'urn:solid-server:default:variable:rootFilePath': '.data' }`.
    *
+   * `settings` corresponds to values that could be passed to the CLI.
+   * E.g.: `{ rootFilePath: '.data' }`.
+   * Shorthand parameter names can not be used, so `{ f: '.data' }` would not work.
+   *
+   * If a certain would set a specific variable,
+   * and both are present in `variableBindings` and `settings`,
+   * the value in `variableBindings` will take priority.
+   *
    * @param loaderProperties - Components.js loader properties.
    * @param configFile - Path to the server config file.
    * @param variableBindings - Bindings of Components.js variables.
+   * @param settings - Settings that need to be resolved.
    */
   public async create(
     loaderProperties: IComponentsManagerBuilderOptions<App>,
     configFile: string,
-    variableBindings: VariableBindings,
+    variableBindings?: VariableBindings,
+    settings?: Settings,
   ): Promise<App> {
-    // Create a resolver to translate (non-core) CLI parameters into values for variables
-    const componentsManager = await this.createComponentsManager<App>(loaderProperties, configFile);
+    const componentsManager = await this.createComponentsManager<any>(loaderProperties, configFile);
+    const cliResolver = await this.createCliResolver(componentsManager);
+
+    const parsedVariables = await this.resolveSettings(cliResolver.settingsResolver, { ...settings });
 
     // Create the application using the translated variable values
-    return componentsManager.instantiate(DEFAULT_APP, { variables: variableBindings });
+    return this.createApp(componentsManager, { ...parsedVariables, ...variableBindings });
```

Comment:
So config variable override CLI params? Unsure here (would add comment in any case)